packages/slider/src/HandleController.ts

```diff
@@ -0,0 +1,540 @@
+        event.stopPropagation();
+        const { input } = this.getHandleElements(model.handle);
+        input.value = this.calculateHandlePosition(event, model).toString();
+        model.handle.value = parseFloat(input.value);
+        this.requestUpdate();
+    }
+
+    /**
+     * Keep the slider value property in sync with the input element's value
+     */
+    private onInputChange(event: Event, model: ModelValue): void {
+        const input = event.target as HTMLInputElement;
+        model.handle.value = input.valueAsNumber;
+
+        this.requestUpdate();
+        this.dispatchChangeEvent(input, model.handle);
+    }
+
+    private onInputFocus(event: Event, model: ModelValue): void {
+        const input = event.target as HTMLInputElement;
+        let isFocusVisible;
+        try {
+            isFocusVisible =
+                input.matches(':focus-visible') ||
+                this.host.matches('.focus-visible');
+            /* c8 ignore next 3 */
+        } catch (error) {
+            isFocusVisible = this.host.matches('.focus-visible');
+        }
+        model.handle.highlight = isFocusVisible;
+        this.requestUpdate();
+    }
+
+    private onInputBlur(model: ModelValue): void {
+        model.handle.highlight = false;
+        this.requestUpdate();
+    }
+
+    private dispatchChangeEvent(
+        input: HTMLInputElement,
+        handle: SliderHandle
+    ): void {
+        input.valueAsNumber = handle.value;
+
+        const changeEvent = new Event('change', {
+            bubbles: true,
+            composed: true,
+        });
+
+        handle.dispatchEvent(changeEvent);
+    }
+
+    /**
+     * Returns the value under the cursor
+     * @param: PointerEvent on slider
+     * @return: Slider value that correlates to the position under the pointer
+     */
+    private calculateHandlePosition(
+        event: PointerEvent | MouseEvent,
+        model: ModelValue
+    ): number {
+        const rect = this.boundingClientRect;
+        const minOffset = rect.left;
+        const offset = event.clientX;
+        const size = rect.width;
+
+        const normalized = (offset - minOffset) / size;
+        const value = model.normalization.fromNormalized(
+            normalized,
+            model.range.min,
+            model.range.max
+        );
+
+        /* c8 ignore next */
+        return this.host.isLTR ? value : model.range.max - value;
+    }
+
+    public renderHandle(
+        model: ModelValue,
+        index: number,
+        zIndex: number
+    ): TemplateResult {
+        const isActive = this.activeHandle === model.name;
+        const classes = {
+            handle: true,
+            dragging: this.draggingHandle?.handleName === model.name,
+            'handle-highlight': model.highlight,
+        };
+        const style = {
+            [this.host.isLTR ? 'left' : 'right']: `${
+                model.normalizedValue * 100
+            }%`,
+            'z-index': zIndex.toString(),
```

Comment:
Just an FYI, but, `styleMap()` does accept JS values for CSS properties, if you wanted.

Reply:
I'm not sure what change you are proposing here (although happy to learn!). 

```javascript
[this.host.isLTR ? 'left' : 'right']: `${
    model.normalizedValue * 100 
}%`,
```
This is a JS value, isn't it? I need the template string to stick the `%` on the end.

Reply:
Not that important, but you can do `zIndex: zIndex.toString(),`, IIRC. So, `zIndex` the JS property for the the property...never mind me.